site/app/libraries/NotebookUtils.php

```diff
@@ -17,13 +17,44 @@ public static function jupyterToSubmittyNotebook(string $filepath): array {
         }
 
         $cells = [];
-
+        // Note: SVG files are not supported due to XSS risks
+        $mime_types = [
+            'image/png',
+            'image/jpeg',
+            'image/gif',
+            'image/bmp',
+            'text/plain', // Fall back to text/plain if it is available.
+        ];
+        $img_size_limit = 1024 * 1024; // 1MB limit
         foreach ($filedata['cells'] as $cell) {
             switch ($cell['cell_type']) {
                 case 'markdown':
+                    $markdown = is_array($cell['source']) ? implode($cell['source']) : (string) $cell['source'];
+                    // Render attachment images (not HTML)
+                    if (isset($cell['attachments']) && count($cell['attachments']) > 0) {
+                        foreach ($cell['attachments'] as $filename => $attachment) {
+                            foreach ($attachment as $mime => $base64) {
+                                $log_message = '';
```

Comment:
Do we actually want to deal with handling an empty string message later?  Perhaps don't include the log message later if there's nothing to show?